src/MongoDBRepository.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace App\Broadway\ReadModel;
+
+use App\MongoDB\Client;
+use Assert\Assertion as Assert;
+use Broadway\ReadModel\ReadModelInterface;
+use Broadway\ReadModel\RepositoryInterface;
+use Broadway\Serializer\SerializerInterface;
+use MongoDB\Model\BSONDocument;
+
+/**
+ * @author Robin van der Vleuten <robin@webstronauts.co>
+ */
+class MongoDBRepository implements RepositoryInterface
+{
+    /**
+     * @var Client
+     */
+    private $client;
+
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
+    /**
+     * @var string
+     */
+    private $collection;
+
+    /**
+     * @var string
+     */
+    private $class;
+
+    /**
+     * Constructor.
+     *
+     * @param Client              $mongodb
+     * @param SerializerInterface $serializer
+     * @param string              $collectionName
+     * @param string              $class
+     */
+    public function __construct(Client $mongodb, SerializerInterface $serializer, string $collectionName, string $class, array $notPersistedFields = [])
```

Comment:
It was a wrapper around the native mongodb client to have support for connection URLs. I'll replace that back with just injecting a native mongodb client through the factory.